bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryListener.java

```diff
@@ -65,5 +68,7 @@
      *            instead
      * @return collection of thing UIDs of all removed things
      */
-    Collection<ThingUID> removeOlderResults(DiscoveryService source, long timestamp, Collection<ThingTypeUID> thingTypeUIDs);
+    @Nullable
```

Comment:
Why do we support a nullable return type? The JavaDoc states that a collection is returned and wouldn't it be simpler if the return value could be used in a for each loop without check for null first?

Reply:
Sure, but then I have to fix at least the wemo binding, hue binding and a tradfri test:

https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoMakerHandler.java#L325

https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoHandler.java#L490

https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoCoffeeHandler.java#L505

https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/handler/HueBridgeHandler.java#L531

https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.tradfri.test/src/test/java/org/eclipse/smarthome/binding/tradfri/discovery/TradfriDiscoveryServiceTest.java#L73

Since in ESH these bindings alone violate the @NonNull I did not put it there to also not break any openHAB bindings.

So my question now is: Shall I fix those ESH bindings within this PR? Create a separate one because its independent from this discovery package? How shall we deal with such situations?